### PR TITLE
Fix New Jersey property tax credit income eligibility logic

### LINKED ISSUE #4475: Fix New Jersey property tax credit income eligibility logic
A number of bugs introduced in PR #4461 have caused a massive regression in NJ income tax testing results.  Income eligibility for the NJ property tax credit is based on NJ gross income (not federal AGI as misspecified in PR #4461).   Here are examples of the recently introduced bugs in the NJ income tax code:

https://github.com/PolicyEngine/policyengine-us/blob/e3edf3fe1f3597fc2103c1a00571e4962d6e37c1/policyengine_us/variables/gov/states/nj/tax/income/credits/property_tax_credit/nj_property_tax_credit_eligible.py#L40-L42

https://github.com/PolicyEngine/policyengine-us/blob/e3edf3fe1f3597fc2103c1a00571e4962d6e37c1/policyengine_us/parameters/gov/states/nj/tax/income/credits/property_tax/income_limit.yaml#L1

### PR DESCRIPTION
**Pull request by hua7450**
Fixes #4475

### CODE REVIEW

File: policyengine_us/variables/gov/states/nj/tax/income/credits/property_tax_credit/nj_property_tax_credit_eligible.py [hunk 1]
```diff
@@ -37,14 +37,8 @@ def formula(tax_unit, period, parameters):
             | senior_head_or_spouse
         )
 
-        federal_agi = tax_unit("adjusted_gross_income", period)
-        filing_status = tax_unit("filing_status", period)
-        agi_eligible = federal_agi <= p.income_limit[filing_status]
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/nj/tax/income/credits/property_tax_credit/nj_property_tax_credit_eligible.py:42]: Does this mean that we need to remove the parameter file or do we use it elsewhere?

Comment (hua7450) [reply policyengine_us/variables/gov/states/nj/tax/income/credits/property_tax_credit/nj_property_tax_credit_eligible.py:42]: We use it for `nj_property_tax_deduction_eligible`

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/nj/tax/income/credits/property_tax_credit/nj_property_tax_credit_eligible.py:42]: Ok makes sense, so these are the only changes we need?

Comment (hua7450) [reply policyengine_us/variables/gov/states/nj/tax/income/credits/property_tax_credit/nj_property_tax_credit_eligible.py:42]: Yes

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**